Sources/Core/API/Texture.swift

```diff
@@ -13,16 +13,20 @@ public struct Texture {
     public let name: String
     /// Any pre-loaded image that should make up the texture
     public let image: Image?
+    /// The format of the texture to load
+    public let format: TextureFormat
 
-    /// Initialize a texture with the name of a bundled image to load
-    public init(name: String) {
+    /// Initialize a texture with the name and the format of a bundled image to load
+    public init(name: String, format: TextureFormat = .png) {
         self.name = name
         self.image = nil
+        self.format = format
     }
 
     /// Initialize a texture with a pre-loaded image
-    public init(image: Image) {
+    public init(image: Image, format: TextureFormat = .unknown) {
```

Comment:
Since `format` is not really relevant for pre-loaded images, I suggest skipping this parameter. Instead of `unknown`, how about making `format` an optional instead, and letting it be `nil` for pre-loaded textures?